staging/src/k8s.io/apiserver/pkg/authentication/authenticator/metrics.go

```diff
@@ -75,21 +76,43 @@ func init() {
 	legacyregistry.MustRegister(authenticationLatency)
 }
 
-func recordAuthMetrics(resp *authenticator.Response, ok bool, err error, apiAudiences authenticator.Audiences, authStart time.Time) {
-	var resultLabel string
+// ResetMetrics resets all auth-related metrics - intended for use in unit tests only.
+// Since prometheus' gatherer is global, other tests may have updated metrics already, so
+// we need to reset them prior running unit tests.
+// This also implies that we can't run metrics tests in parallel with other auth tests.
+func ResetMetrics() {
+	authenticatedUserCounter.Reset()
+	authenticatedAttemptsCounter.Reset()
+}
+
+func recordAuthMetrics(ctx context.Context, resp *Response, authSuccess bool, err error, authStart time.Time) {
+	var (
+		resultLabel       string
+		authenticatorName = "unknown"
+		instrumentedError *InstrumentedError
+	)
+	// Ignoring "ok" since we expect nil value.
+	apiAudiences, _ := AudiencesFrom(ctx)
+
+	if resp != nil && resp.AuthenticatorName != "" {
+		authenticatorName = resp.AuthenticatorName
+	}
 
 	switch {
 	case err != nil || (resp != nil && !audiencesAreAcceptable(apiAudiences, resp.Audiences)):
 		resultLabel = errorLabel
-	case !ok:
+		if errors.As(err, &instrumentedError) && instrumentedError.AuthenticatorName != "" {
```

Comment:
Pretty sure this is passing in a `**InstrumentedError`?

Reply:
Removed the InstrumentedError type.